src/node_api.cc

```diff
@@ -46,26 +47,34 @@ void node_napi_env__::CallFinalizer(napi_finalize cb, void* data, void* hint) {
 
 template <bool enforceUncaughtExceptionPolicy>
 void node_napi_env__::CallFinalizer(napi_finalize cb, void* data, void* hint) {
-  if (destructing) {
-    // we can not defer finalizers when the environment is being destructed.
-    v8::HandleScope handle_scope(isolate);
-    v8::Context::Scope context_scope(context());
-    CallbackIntoModule<enforceUncaughtExceptionPolicy>(
-        [&](napi_env env) { cb(env, data, hint); });
-    return;
+  v8::HandleScope handle_scope(isolate);
+  v8::Context::Scope context_scope(context());
+  CallbackIntoModule<enforceUncaughtExceptionPolicy>(
+      [&](napi_env env) { cb(env, data, hint); });
+}
+
+void node_napi_env__::EnqueueFinalizer(v8impl::RefTracker* finalizer) {
+  napi_env__::EnqueueFinalizer(finalizer);
+  if (!finalization_scheduled) {
```

Comment:
We must also check that we are not `destructing == true` because we do not want the `env` ref count bounce from zero in case if are called from the `node_napi_env__` while executing `FinalizeAll`.

Reply:
Thank you for pointing this out. Updated.